segmentstore/server/src/main/java/io/pravega/segmentstore/server/AttributeSerializer.java

```diff
@@ -34,6 +34,8 @@
      * @throws IOException If an exception occurred.
      */
     public static void serialize(Map<UUID, Long> attributes, DataOutputStream stream) throws IOException {
+        Preconditions.checkArgument(attributes.size() <= Short.MAX_VALUE,
```

Comment:
No. This checks for absolute maximum that can be serialized, which is Short.MAX_VALUE. AttributeSerializer deals with serialization, not with the hard limit we enforce in the SegmentStore. 

Consider the following scenario: in the future we decide to decrease the maximum number of attributes allowed. A Segment that's activated (even as sealed), which previously had a large number of attributes, would now be unable to serialize into the MetadataCheckpoint because the new attribute limit would be too small. Not a situation you want to get into.